src/shared/utils/fluxFunctions.ts

```diff
@@ -0,0 +1,117 @@
+import {FluxFunction} from 'src/types/shared'
+import {CLOUD} from 'src/shared/constants'
+import {FROM, UNION} from 'src/shared/constants/fluxFunctions'
+import {isFlagEnabled} from 'src/shared/utils/featureFlag'
+
+export const isPipeTransformation = (func: FluxFunction) => {
+  if (CLOUD && isFlagEnabled('fluxDynamicDocs')) {
+    return func.fluxType.startsWith('<-', 1)
+  }
+  return !['from', 'union'].includes(func.name)
+}
+
+export const functionRequiresNewLine = (funcName: string): boolean => {
+  switch (funcName) {
+    case FROM.name:
+    case UNION.name: {
+      return true
+    }
+    default:
+      return false
+  }
+}
+
+export enum InjectionType {
+  OnOwnLine = 'onOwnLine',
+  SameLine = 'sameLine',
+}
+
+export interface InjectionPosition {
+  row: number
+  column: number
+  shouldStartWithNewLine: boolean
+  shouldEndInNewLine: boolean
+}
+
+export function calcInjectionPosition(
```

Comment:
this function is exactly the same, only extracted to be used with any monaco-editor (not just the editor.context used by notebooks).